### PR TITLE
Run all workflows at 7am (UTC) everyday; Add paths and paths-ignore

### PR DESCRIPTION
**Pull request by wuweicaifb**
## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

1. Add nightly check with all workflows except deploy-gh-pages. The workflow will be triggered every midnight at PST (7 am UTC).
2. Skip tests, pyre, and lints for PR that only includes doc changes; Only build doc page when docs or docusaurus is updated.

## Test Plan

Let's see what will happen this midnight!

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
   push:
     branches: [main]
   pull_request:
-    branches:
-      - "*"
+    paths-ignore:
+      - 'docs/**'
+      - 'docusaurus/**'
+  schedule:
+    - cron: '0 7 * * *'
```
Comment (dbeylkin) [inline .github/workflows/lint.yml:8]: Who gets notified if this fails since it's not tied to a PR?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -1,13 +1,14 @@
 name: Lint
 
 on:
-  # Trigger the workflow on push or pull request,
-  # but only for the main branch
   push:
     branches: [main]
   pull_request:
-    branches:
-      - "*"
```
Comment (dbeylkin) [inline .github/workflows/lint.yml:10]: Why is this deleted? Is this the default if not specified?

**Review by dbeylkin — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+    paths-ignore:
```
Comment (dbeylkin) [inline .github/workflows/lint.yml]: For linter, do we still ignore these paths? I thought it might still catch white space issues for example.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (wuweicaifb) [reply .github/workflows/lint.yml:10]: Not exactly. Actually we need to remove it or change it to `**`. Per github action doc (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet), `*` match zero or more characters except "/", but `**` match everything. And you're right, if branches is not specified, there will be no filter on it.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (wuweicaifb) [reply .github/workflows/lint.yml:8]: Yes. We get notified whenever an action fails, no matter how it is triggered.

Comment (dbeylkin) [reply .github/workflows/lint.yml:8]: Who is we in this case? Anyone watching the repo? Contributors?

Comment (wuweicaifb) [reply .github/workflows/lint.yml:8]: Who is watching the repo:

https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#github-actions-notification-options
